Source/WTF/wtf/text/StringImpl.h

```diff
@@ -1152,6 +1156,36 @@ inline void StringImpl::copyCharacters(DestinationCharacterType* destination, co
         }
         memcpy(destination, source, numCharacters * sizeof(DestinationCharacterType));
     } else {
+#if CPU(ARM64)
+        // SIMD Upconvert.
+        if constexpr (std::is_same_v<SourceCharacterType, LChar> && std::is_same_v<DestinationCharacterType, UChar>) {
+            const auto* const end = destination + numCharacters;
+            constexpr uintptr_t memoryAccessSize = 64;
+
+            if (numCharacters >= memoryAccessSize) {
+                constexpr uintptr_t memoryAccessMask = memoryAccessSize - 1;
+                const auto* const simdEnd = destination + (numCharacters & ~memoryAccessMask);
```

Comment:
So simdEnd = ((destination + numCharacters) / memoryAccessSize) * memoryAccessSize, right? Can clang not do this transformation for us, or am I getting the math wrong?

Reply:
It is slightly different since destination is `UChar*` pointer (UChar is 2 byte size).
We need to round down numCharacters to N * memoryAccessSize because this memoryAccessSize is based on the source's one. And I think this bit mask thing is easier to read for that purpose (and it is aligned to the existing code in ASCIIFastPath.h).

Reply:
Not sure the second `const` makes it easier to read.

Reply:
Dropped.